### PR TITLE
Adds the T series pistol/revolver/shotgun/autoshotgun and some stuff

### PR DESCRIPTION
**Pull request by Kuro-ai**
## About The Pull Request

Adds the reworked pistol/revolver/pump shotgun/auto shotgun with some new sounds,sprites and balance changes as well as fixing some pathing on ammo boxes

## Why It's Good For The Game

and now the T series is even more complete

## Changelog
:cl:
add: Reworked pistol/revolver/auto shotgun/pump shotgun
tweak: tweaked a few type paths
balance: balance changes on the 4 guns
soundadd: added a new sound for pistol shots and pump shotgun shots
imageadd: added new sprites for the 4 guns
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -7. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -564,3 +564,41 @@
...
+						)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_LOAD_INTO_CHAMBER
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC)
+	starting_attachment_types = list(/obj/item/attachable/stock/tx15)
+	attachable_offset = list("muzzle_x" = 30, "muzzle_y" = 16,"rail_x" = 12, "rail_y" = 17, "under_x" = 26, "under_y" = 15, "stock_x" = 26, "stock_y" = 13)
+	gun_skill_category = GUN_SKILL_SHOTGUNS
+
+	fire_delay = 1.1 SECONDS
+	burst_amount = 1
```
Comment (spookydonut) [inline code/modules/projectiles/guns/rifles.dm]:
```suggestion
	burst_amount = 1

```

File: code/modules/projectiles/magazines/rifles.dm [hunk 2]
```diff
@@ -198,3 +198,23 @@
...
+	gun_type = /obj/item/weapon/gun/rifle/standard_autoshotgun
+
+/obj/item/ammo_magazine/rifle/tx15_slug
+	name = "\improper TX-15 slug magazine (16 gauge)"
+	desc = "A magazine of 16 gauge slugs, for the TX-15."
+	caliber = "16 gauge"
+	icon_state = "tx15_slug"
+	default_ammo = /datum/ammo/bullet/shotgun/tx15_slug
+	max_rounds = 12
+	gun_type = /obj/item/weapon/gun/rifle/standard_autoshotgun
```
Comment (spookydonut) [inline code/modules/projectiles/magazines/rifles.dm]:
```suggestion
	gun_type = /obj/item/weapon/gun/rifle/standard_autoshotgun

```

**Comment by TheBimmer231** [comment]
huh, double approved since my connection died while doing it

**Comment by Kuro-ai** [comment]
Lmao,nice moves

File: code/game/objects/items/storage/belt.dm [hunk 3]
```diff
@@ -569,15 +569,32 @@
...
+	name = "\improper T457 pattern TP-44 holster rig"
+	desc = "The T457 is the standard load-bearing equipment of the TGMC. It consists of a modular belt with various clips. This version is for the TP-44 Revolver."
+	icon_state = "tp44_holster"
+	item_state = "tp44_holster"
 	can_hold = list(
-		/obj/item/weapon/gun/revolver,
-		/obj/item/ammo_magazine/revolver
+		/obj/item/weapon/gun/revolver/standard_revolver,
+		/obj/item/ammo_magazine/revolver/standard_revolver
 		)
```
Comment (Rohesie) [inline code/game/objects/items/storage/belt.dm]: Can we keep the item?

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -650,6 +652,40 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 
+/datum/ammo/bullet/shotgun/tx15_flechette
+	name = "shotgun flechette shell"
+	icon_state = "flechette"
+	hud_state = "shotgun_flechette"
+	bonus_projectiles_type = /datum/ammo/bullet/shotgun/tx15_flechette/spread
+	bonus_projectiles_amount = 4
+	bonus_projectiles_scatter = 4
+	accuracy_var_low = 7
+	accuracy_var_high = 7
```
Comment (Rohesie) [inline code/modules/projectiles/ammo_datums.dm]: Why accuracy variation when accuracy is already a % roll ahhh

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -650,6 +652,40 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	max_range = 15
+	damage = 20
+	damage_falloff = 0.25
+	penetration = 15
+
+/datum/ammo/bullet/shotgun/tx15_flechette/spread
+	name = "additional flechette"
+	icon_state = "flechette"
+	accuracy_var_low = 7
+	accuracy_var_high = 7
```
Comment (Rohesie) [inline code/modules/projectiles/ammo_datums.dm]: Same on accuracy variation. Nuke this.

File: code/modules/projectiles/guns/pistols.dm [hunk 6]
```diff
@@ -45,6 +45,28 @@
...
+	caliber = "9x19mm Parabellum" //codex
+	max_shells = 14 //codex
+	fire_sound = 'sound/weapons/guns/fire/tp14.ogg'
+	reload_sound = 'sound/weapons/guns/interact/tp14_reload.ogg'
+	current_mag = /obj/item/ammo_magazine/pistol/standard_pistol
+
+	attachable_offset = list("muzzle_x" = 29, "muzzle_y" = 20,"rail_x" = 13, "rail_y" = 23, "under_x" = 19, "under_y" = 13, "stock_x" = 21, "stock_y" = 17)
+
+	fire_delay = 0.125 SECONDS
+	accuracy_mult = 1.00
```
Comment (Rohesie) [inline code/modules/projectiles/guns/pistols.dm]:
```suggestion
	accuracy_mult = 1
```
Avoid extra zeros. They make mass replacements more annoying.

File: code/modules/projectiles/guns/revolvers.dm [hunk 7]
```diff
@@ -295,6 +295,36 @@
...
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/heavy_barrel,
+						/obj/item/attachable/quickfire,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/compensator,
+						/obj/item/attachable/lasersight)
+	attachable_offset = list("muzzle_x" = 34, "muzzle_y" = 19,"rail_x" = 15, "rail_y" = 23, "under_x" = 22, "under_y" = 14, "stock_x" = 22, "stock_y" = 19)
+	fire_delay = 0.3
+	accuracy_mult_unwielded = 0.85
+	accuracy_mult = 1.00
```
Comment (Rohesie) [inline code/modules/projectiles/guns/revolvers.dm]:
```suggestion
	accuracy_mult = 1
```

**Comment by David-Stormwell** [comment]
Last I checked today revolution ammo is still M44 not T44, and won't go in, the speed loaders that is.

**Comment by Rohesie** [comment]
Once that bug is fixed this should be ready for merge.

**Comment by Kuro-ai** [comment]
> Once that bug is fixed this should be ready for merge.

solved

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-8). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**